src/common/VirtualTerminal.cpp

```diff
@@ -155,14 +185,24 @@ namespace SDDM {
 
             // fallback to active VT
             if (vt <= 0) {
+#ifdef __FreeBSD__
+                int vtActive = 0;
+                if (ioctl(fd, VT_GETACTIVE, &vtActive) < 0) {
+                    qCritical() << "Failed to get current VT:" << strerror(errno);
+                    close(fd);
```

Comment:
Same goes for here, the qScopeGuard added upstream since 0.19.0 means you don’t want this